docs/upload_specifications.ADR.md

```diff
@@ -0,0 +1,170 @@
+fasta_file, gff_file, genome_name, source, release, genetic_code, generate_missing_genes
+FASTA File Path, GFF3 File Path, Metagenome Object Name, Source of metagenome, Release or Version of the Source Data, Genetic Code for protein translation, Spoof Genes for parentless CDS
+mygenome.fa, mygenome.gff3, mygenomeobject, , , 11, 0
+mygenome2.fa, mygenome2.gff3, mygenomeobject2, yermumspoo, 30456, 11, 1
+...
+```
+
+The file, by row, is:
+1. The data type, in this case `gff_metagenome`. This is a data type listed in the
+  [Mappings.py](https://github.com/kbase/staging_service/blob/master/staging_service/autodetect/Mappings.py)
+  file in the StS. The Narrative is expected to understand these types and map them to uploader
+  apps.
+2. The IDs of the app inputs from the `spec.json` file. 
+3. The corresponding human readable names of the app inputs from the `display.yaml` file.
+4. (and beyond) Import specifications. Each line corresponds to a single import.
+
+For Excel files, the first two rows may be hidden in any provided templates. Additionally,
+Excel files may contain multiple data types, one per tab.
+
+As part of this project we will deliver:
+1. CSV templates for each in scope app (e.g. the first 3 lines of the example file)
+2. An Excel template containing a tab for each in scope app
+3. A `README.md` file explaining how to use the templates.
+   * The `README.md` should include a link to rich upload specification documentation on the KBase
+     website once it is developed.
+   * Note: cover booleans which are not intuitive.
+
+These files will reside in the StS repo. As part of the front end effort, some means of
+delivering the templates and instructions to users will be developed.
+
+Currently, for all in scope apps, the inputs are all strings, numbers, or booleans. There are
+no unusual inputs such as grouped parameters or dynamic lookups. Including future upload apps
+with these features in CSV-based upload may require additional engineering.
+
+## User operations
+
+* The user uploads the upload specification files to the staging area along with all the files
+  inluded in the specification. 
+* The user selects the `Upload Specification` type for the specification files.
+  * The user may also select other files in the staging area to include in the upload along
+    with the files listed in the specification.
+    * The user *does not* have to select any files included in the specification.
+* The user clicks `Import Selected`.
+
+As such, a new type must be added to the StS: `upload_specification` with the title
+`Upload Specification`. *Nota bene*: It may be preferable to have the Narrative specify the
+titles to display in the staging area dropdown rather than the StS.
+
+## Narrative Bulk Upload cell operations
+
+When the bulk upload cell sees a `upload_specification` data type, it calls out to the StS
+endpoint detailed below to get the parsed specifications, and then autofills itself with
+those specifications.
```

Comment:
More specifically it would be the data import panel sees the data type upon clicking the "import" button, gets the parsed info from the staging service, then requests the narrative to generate an bulk upload cell based off of those specifications.

Reply:
I removed any references to particular implementation details on the narrative side